pandas/core/indexes/datetimelike.py

```diff
@@ -695,6 +706,23 @@ def __sub__(self, other):
                 return self._sub_datelike(other)
             elif isinstance(other, Period):
                 return self._sub_period(other)
+            elif (isinstance(self, DatetimeIndex) and is_offsetlike(other) and
+                  not isinstance(other, ABCSeries)):
+                # Array of DateOffset objects
+                if len(other) == 1:
+                    return self - other[0]
+                else:
+                    from pandas.errors import PerformanceWarning
+                    warnings.warn("Adding/subtracting array of DateOffsets to "
```

Comment:
Ah, I see that you added this yourself.  Still, feel a little uneasy about this.